### ISSUE TITLE
Aggregate breaking change instructions in CHANGELOG

### ISSUE THREAD

**Issue by arboleya** [FuelLabs/fuels-ts#2609]
We currently use a digest to signal all PRs for new releases:
 - https://github.com/FuelLabs/fuels-ts/releases/tag/v0.90.0

At the top, we have a section grouping all breaking changes with links to related PRs. Then, inside each PR, one can find detailed instructions about the breaking change it introduced.

The idea is to improve our CHANGELOG automation so that instructions from every PR are always included in the release notes, allowing users to find everything they need in a single place.

> **TODO**: Draft template of how it will look like.

**Comment by arboleya** [comment]
Blocked by:
 - https://github.com/FuelLabs/fuels-ts/issues/2523

**Comment by calldelegation** [comment]
@arboleya the GOAT 🙏 thank you

**Comment by nedsalk** [comment]
In relation to the implementation, we'd have to do the following two:
1. Have a CI check to verify a breaking change PR contains a section in which the breaking change is explained
2. On changelog generation, fetch the PR's description and copy that section over to the changelog

@arboleya the breaking change description on the PR would contain everything necessary to complete this issue, so how is this issue being blocked by #2523 which is concerned with issue creation?

**Comment by arboleya** [comment]
I will create a default/standard template for the PRs with a pre-defined breaking change section.

**Comment by SilentCicero** [comment]
@arboleya if you can also include a small section which describes the changes from a high level, i.e. a description of the update:
In this update we did X, Y, Z to improve A, B and C.

A breif description we can pull in for the update in plain english will be helpful. It can be short, but just describe what the team was up to when they updated for the version.

The other Chores and Migration Notes section I see listed in the lower comments here seems perfect for what we need.

**Comment by arboleya** [comment]
@SilentCicero Do you mean a high-level description summarizing what is **_relevant_** about the PR?

Does it need to be fluid text, or can it still be bullet points or something similar?

These Release Notes are an automated amalgamation of individual PRs, so this info must still come from the PRs. There could be an optional section that people can fill out on each PR, provided their PR is relevant enough. Then, we can pull and combine those at the top before the digest begins — would that work?

If a given release doesn't have relevant notes, we could have a default one:
 - General bug fixes and improvements
